cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -345,6 +349,15 @@ protected void configure() {
       // Used in InMemoryProgramRunDispatcher, TetheringClientHandler
       install(RemoteAuthenticatorModules.getDefaultModule(TetheringAgentService.REMOTE_TETHERING_AUTHENTICATOR,
                                                           Constants.Tethering.CLIENT_AUTHENTICATOR_NAME));
+      install(
+        new FactoryModuleBuilder()
```

Comment:
Here we took the reference of `InMemoryConfigurator` and `RemoteConfigurator`  in above: basically we need both inmemory and remote implementation of the operation runner that does push/list/pull, does it make sence?

Reply:
You have a service as a coordination layer that uses Guice to create / pass things between the manager and runner. It makes it very complex and hard to track.
You are doing `sourceControlFactory.create(createRepoManager(namespace)).push(appsToPush, commitMeta)`.
If you'll merge two layers or move manager creation  down it will be much simper. 
Ideally it may even look like `push(namespace, appsToPush, commitMeta`. Or even without a namespace as it's part of appsToPush

Reply:
@tivv will move down the creation of RepoManager into operation runner

Reply:
@tivv Moved down the creation of RepoManager, still had to pass in the RepoConfig because it's now only accessible from the service layer. Should we add a store layer like `RepositoryStore` that handles repoTable and called from the service/runner?

Reply:
Well, what other layer do we have? We have handlers and services calling each other. This factory is a service. It's nice if it can be created by Guice, but it's not a requirement.